### ISSUE TITLE
Failed to load format Error: Trying to leave Layout {"endData":{"rootNode":1,"root":[0,0,0,0,0,0,0,0]}} when stack is empty

### ISSUE THREAD

**Issue by chalvo55** [jlfwong/speedscope#221]
I get this error when loading from `chrome://tracing`. Happy to provide you with the JSON privately.

**Comment by jlfwong** [comment]
@chalvo55 Yes please! You can email a trace to jamie.lf.wong@gmail.com if you'd prefer it be private.

**Comment by chalvo55** [comment]
@jlfwong sent you an email, thanks! :)

**Comment by jlfwong** [comment]
I took a look at the profile sent via email, and it seems like the offending portion of the trace can be reduced down to the first 38 events.

Here's a reduced repro case which I'm pretty confident doesn't contain any confidential information.

```
[{"tdur":101,"name":"KeyedServiceFactory::GetServiceForContext","tts":789696,"pid":20870,"ts":0,"cat":"browser,startup","tid":775,"dur":186,"ph":"B"},{"tdur":101,"name":"KeyedServiceFactory::GetServiceForContext","tts":789696,"pid":20870,"ts":186,"cat":"browser,startup","tid":775,"dur":186,"ph":"E"},{"tdur":14,"name":"KeyedServiceFactory::GetServiceForContext","tts":790255,"pid":20870,"ts":643,"cat":"browser,startup","tid":775,"dur":15,"ph":"B"},{"tdur":14,"name":"KeyedServiceFactory::GetServiceForContext","tts":790255,"pid":20870,"ts":658,"cat":"browser,startup","tid":775,"dur":15,"ph":"E"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":790281,"pid":20870,"ts":670,"cat":"browser,startup","tid":775,"dur":7,"ph":"B"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":790281,"pid":20870,"ts":677,"cat":"browser,startup","tid":775,"dur":7,"ph":"E"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":790299,"pid":20870,"ts":688,"cat":"browser,startup","tid":775,"dur":7,"ph":"B"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":790299,"pid":20870,"ts":695,"cat":"browser,startup","tid":775,"dur":7,"ph":"E"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":790314,"pid":20870,"ts":703,"cat":"browser,startup","tid":775,"dur":8,"ph":"B"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":790314,"pid":20870,"ts":711,"cat":"browser,startup","tid":775,"dur":8,"ph":"E"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":790348,"pid":20870,"ts":737,"cat":"browser,startup","tid":775,"dur":8,"ph":"B"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":790348,"pid":20870,"ts":745,"cat":"browser,startup","tid":775,"dur":8,"ph":"E"},{"tdur":20,"name":"KeyedServiceFactory::GetServiceForContext","tts":791575,"pid":20870,"ts":3389,"cat":"browser,startup","tid":775,"dur":24,"ph":"B"},{"tdur":20,"name":"KeyedServiceFactory::GetServiceForContext","tts":791575,"pid":20870,"ts":3413,"cat":"browser,startup","tid":775,"dur":24,"ph":"E"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":791614,"pid":20870,"ts":3422,"cat":"browser,startup","tid":775,"dur":17,"ph":"B"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":791614,"pid":20870,"ts":3439,"cat":"browser,startup","tid":775,"dur":17,"ph":"E"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":791631,"pid":20870,"ts":3448,"cat":"browser,startup","tid":775,"dur":8,"ph":"B"},{"tdur":8,"name":"KeyedServiceFactory::GetServiceForContext","tts":791631,"pid":20870,"ts":3456,"cat":"browser,startup","tid":775,"dur":8,"ph":"E"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":791648,"pid":20870,"ts":3465,"cat":"browser,startup","tid":775,"dur":7,"ph":"B"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":791648,"pid":20870,"ts":3472,"cat":"browser,startup","tid":775,"dur":7,"ph":"E"},{"tdur":9,"name":"KeyedServiceFactory::GetServiceForContext","tts":791664,"pid":20870,"ts":3480,"cat":"browser,startup","tid":775,"dur":11,"ph":"B"},{"tdur":9,"name":"KeyedServiceFactory::GetServiceForContext","tts":791664,"pid":20870,"ts":3491,"cat":"browser,startup","tid":775,"dur":11,"ph":"E"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":791680,"pid":20870,"ts":3497,"cat":"browser,startup","tid":775,"dur":7,"ph":"B"},{"tdur":7,"name":"KeyedServiceFactory::GetServiceForContext","tts":791680,"pid":20870,"ts":3504,"cat":"browser,startup","tid":775,"dur":7,"ph":"E"},{"tdur":6,"name":"KeyedServiceFactory::GetServiceForContext","tts":791695,"pid":20870,"ts":3512,"cat":"browser,startup","tid":775,"dur":6,"ph":"B"},{"tdur":6,"name":"KeyedServiceFactory::GetServiceForContext","tts":791695,"pid":20870,"ts":3518,"cat":"browser,startup","tid":775,"dur":6,"ph":"E"},{"tdur":18,"name":"KeyedServiceFactory::GetServiceForContext","tts":791708,"pid":20870,"ts":3525,"cat":"browser,startup","tid":775,"dur":17,"ph":"B"},{"tdur":18,"name":"KeyedServiceFactory::GetServiceForContext","tts":791708,"pid":20870,"ts":3542,"cat":"browser,startup","tid":775,"dur":17,"ph":"E"},{"tdur":16,"name":"KeyedServiceFactory::GetServiceForContext","tts":791905,"pid":20870,"ts":4049,"cat":"browser,startup","tid":775,"dur":18,"ph":"B"},{"tdur":16,"name":"KeyedServiceFactory::GetServiceForContext","tts":791905,"pid":20870,"ts":4067,"cat":"browser,startup","tid":775,"dur":18,"ph":"E"},{"tdur":33,"name":"KeyedServiceFactory::GetServiceForContext","tts":793864,"pid":20870,"ts":7752,"cat":"browser,startup","tid":775,"dur":62,"ph":"B"},{"tdur":33,"name":"KeyedServiceFactory::GetServiceForContext","tts":793864,"pid":20870,"ts":7814,"cat":"browser,startup","tid":775,"dur":62,"ph":"E"},{"name":"UpdateLayoutTree","tts":209004,"pid":20910,"ts":25191,"cat":"blink,devtools.timeline","tid":775,"ph":"E"},{"name":"Layout","tts":209295,"pid":20910,"ts":25710,"cat":"devtools.timeline","tid":775,"ph":"B"},{"tdur":15,"name":"KeyedServiceFactory::GetServiceForContext","tts":797187,"pid":20870,"ts":30850,"cat":"browser,startup","tid":775,"dur":17,"ph":"B"},{"tdur":15,"name":"KeyedServiceFactory::GetServiceForContext","tts":797187,"pid":20870,"ts":30867,"cat":"browser,startup","tid":775,"dur":17,"ph":"E"},{"tdur":14,"name":"KeyedServiceFactory::GetServiceForContext","tts":799452,"pid":20870,"ts":36787,"cat":"browser,startup","tid":775,"dur":17,"ph":"B"},{"tdur":14,"name":"KeyedServiceFactory::GetServiceForContext","tts":799452,"pid":20870,"ts":36804,"cat":"browser,startup","tid":775,"dur":17,"ph":"E"},{"name":"Layout","tts":216109,"pid":20910,"ts":43088,"cat":"devtools.timeline","tid":775,"ph":"E"}]
```

As far as I can tell, this is just malformed. There are more "E" events (20) than "B" events (19), meaning that scopes were closed that were never opened, so it's pretty unclear how to correctly resolve this.

That said, it seems like `chrome://tracing` does import the trace successfully. I'll keep poking around and see what `chrome://tracing` does in this situation.

**Comment by jlfwong** [comment]
@chalvo55 This should be fixed in speedscope@1.5.1, and the fix should be deployed to https://www.speedscope.app/ too

**Comment by chalvo55** [comment]
@jlfwong This is excellent, thanks!
